js/web/lib/wasm/jsep/webgpu/ops/attention.ts

```diff
@@ -652,11 +652,27 @@ export const applyAttention = (
   parameters: AttentionParameters,
   attributes: AttentionAttrs,
 ) => {
+  const pastSequenceLength =
+    parameters.kvNumHeads !== undefined || context.outputCount > 1 ? parameters.pastSequenceLength : 0;
+
+  // context.outputCount comes from KernelOp and is the number of outputs the op has.
+  // If they are not consumed we need to make sure the shaders don't generate the output
+  // since there is no buffer for it.
+  // We check by requesting the output and if not there we'll adjust context.outputCount
+  const presentKeyShape = [
+    parameters.batchSize,
```

Comment:
This shape only works for MHA and GQA.
Attention output 1 shape is [2, B, N, T, H] instead of [B, N, T, H], since it concatenates present_key and present_value as present output.

I think here need extra code like
```
if ( _past === undefined) {// it is MHA or GQA op
   // insert 2 at the beginning of present shape.
} else { // Attention op
}
```

Reply:
Also need handle another special case for GQA that past and present share buffers. In that case, the length is max sequence length.

So probably can just handle MHA here.
```
if (parameters.kvNumHeads === undefined && context.outputCount > 2 ) { // MHA
 ...
}
```